src/Transport/Topology/Topologies/StandardTopology.cs

```diff
@@ -90,8 +90,11 @@ public Func<ICreateQueues> GetQueueCreatorFactory()
 
         public Func<CriticalError, IPushMessages> GetMessagePumpFactory()
         {
-            // todo, get criticial error integrated
-            return error => container.Resolve<IPushMessages>();
+            return error =>
+            {
+                container.Register<CriticalError>(() => error);
```

Comment:
@yvesgoeleven we've discussed this one, but I'm not convinced if changing this would be a good idea. To get that working, we'd need to cast, and that means if another implementation of `IPushMessages` is registered, this code will not work as expected.

```c#
var pump = container.Resolve<IPushMessages>()
(pump as MessagePump)?.SetCriticalError(error); // <-- issue
return pump;```